File: java/Kode/Bow.java
public class Bow extends Weapon {
    private int arrows;
    private double range;
    private double drawSpeed;

    public Bow(String name, int level, String tier, WeaponMaterial material, 
               int arrows, double range, double drawSpeed) {
        // Panggilan ke constructor Base Class (Weapon)
        super(name, level, tier, material);
        this.arrows = arrows;
        this.range = range;
        this.drawSpeed = drawSpeed;
    }

    @Override
    protected String specificsInfo() {
        return "Arrows: " + arrows + ", Range: " + range + ", Draw Speed: " + drawSpeed;
    }

    // Getter untuk kebutuhan Hybrid (Simulasi Multiple Inheritance)
    public int getArrows() { return arrows; }
    public double getRange() { return range; }
    public double getDrawSpeed() { return drawSpeed; }
}

File: cpp/Kode/Bow.cpp
#include "Bow.h"
#include "WeaponMaterial.h" 

Bow::Bow(std::string n, int l, std::string t, const std::vector<std::pair<std::string, double>>& s, 
    std::unique_ptr<WeaponMaterial> m,
    int arr, double rng, double ds) : 
    Weapon(n, l, t, s, std::move(m)), arrows(arr), Range(rng), DrawSpeed(ds) {}

void Bow::displayWeapon() const {
    Weapon::displayWeapon(); 
    std::cout << "  Specifics: Arrows: " << arrows << ", Range: " << Range << ", Draw Speed: " << DrawSpeed << std::endl;
}

void Bow::displayWeaponWithoutMaterial() const {
    Weapon::displayWeaponWithoutMaterial(); 
    std::cout << "  Specifics: Arrows: " << arrows << ", Range: " << Range << ", Draw Speed: " << DrawSpeed << std::endl;
}